### ISSUE TITLE
Errors with Generics - missing trait bounds.

### ISSUE THREAD

**Issue by drahnr** [dtolnay/thiserror#112]
During cleaning up some of the error handling in paritytech/substrate I encountered that using `#[error("yada {0:?}")]` for something along the lines of `struct Erroris<T>(pub Wrapper<T>)` does not enforce a `T: Debug` bound on the underlying generic type, but requires adding a bound on the struct declaration/definition rather than the generated `fmt::Debug` impl. That goes against what is stated as a best practice in the future proofing guide https://rust-lang.github.io/api-guidelines/future-proofing.html?#data-structures-do-not-duplicate-derived-trait-bounds-c-struct-bounds .

https://github.com/paritytech/substrate/pull/7631#discussion_r532700450

It would be awesome if this could be addressed :heart:

**Comment by bkchr** [comment]
Probably a duplicate of: https://github.com/dtolnay/thiserror/issues/79

**Comment by dtolnay** [comment]
Yeah this is tracked in #79, and is just waiting on a PR.
